File: axu4976_lab01/axu4976_lab01/axu4976_lab01.py
# Ashwanthika Umasankar
#UTA ID - 1001854976
#Python version used - Python 3.9.13
#OS -  Windows

import os #import the OS module

#Function that computes the total size of its dorectory an its contents while taking in a parameter 
def directorySize(folder,flag):
    #Setting the initial size to be returned at a later point as 0
    dirSize = 0 
    baseDirSize=0
    if flag==0:
        for var in os.scandir(folder):
            if var.is_file():
                baseDirSize += var.stat().st_size 
        print("\nThe size of files present in base directory:",baseDirSize )
    for var in os.scandir(folder):
        if var.is_file():
            dirSize += var.stat().st_size # Add the size of the file to the size
        elif var.is_dir():
            subdir_size = directorySize(var.path, flag+1) # Recursively calculate the size of the subdirectory
            #print(f"\n Size of subdirectory located at the path {var.path}: {subdir_size} bytes\n") # Print the size of the subdirectory along with its path
            dirSize += subdir_size # Add the size of the subdirectory to the  size

    return dirSize
#get the current directory
def currentDirectory():
    return os.path.dirname(os.path.abspath(__file__))

def main():
    current_directory = currentDirectory()
    total_size = directorySize(current_directory,0) #get the current directory and pass it to the function and compute the total size of directory and subdirectories

    if total_size == 0:
        print("The directory is empty and the total size cannot be found")
    else:
        print(f"\nTotal size of the entire directory is: {total_size} bytes\n")

if __name__ == "__main__":
    main()

File: axu4976_lab01/axu4976_lab01/axu4976_lab01.java
//Ashwanthika Umasankar
//UTA ID - 1001854976
//java version "19.0.2"
//OS -  Windows



import java.io.File;

public class axu4976_lab01 {
    //function directorysize used to compute the total size of a directory and its contents
    //function takes a file object input and returns a long value representing the totalsize using recursion 
    private static long directorySize(File folder,  int flag) {
        long directorySize = 0;
        long baseDirSize = 0;
        //set flag as 0 and compute the size of files belonging oto base dir
        if (flag == 0) {
        File[] fileArray = folder.listFiles();
            if (fileArray != null) {
                int count = fileArray.length;
                int var = 0;
                while (var < count) {
                    File file = fileArray[var];
                    if (file.isFile()) {
                        baseDirSize += file.length();
                    }
                    var++;
                }
            }
        System.out.println("\nThe size of files present in the base directory: " + baseDirSize);
        }
        //add the size and recursively comute subdirectory size
        File[] fileArray  = folder.listFiles();
        if (fileArray != null) {
            int count = fileArray.length;
            int var = 0;
            while (var < count) {
                File file = fileArray[var];
                if (file.isFile()) {
                    directorySize += file.length();
                } else if (file.isDirectory()) {
                    directorySize += directorySize(file, flag+1);
                }
                var++;
            }
        }

        return directorySize;
    }

    public static void main(String[] args) {
        File currentDirectory = new File("."); 
        long totalSize = directorySize(currentDirectory, 0); //get the current directory and pass it to the function and compute the total size of directory and subdirectories

        if (totalSize == 0) {
            System.out.println("The directory is empty and the total size cannot be found");
        } else {
            System.out.println("Total size of the entire directory is " + totalSize + " bytes");
        }
    }
}